src/libertem/io/dataset/hdf5.py

```diff
@@ -266,23 +333,81 @@ def _preprocess(self, tile_data, tile_slice):
             return
         self._corrections.apply(tile_data, tile_slice)
 
+    def _get_reshaped_tile_slice(self, tile_slice):
+        nav_dim = (tile_slice.origin[0] * self.meta.shape[1]) + tile_slice.origin[1]
+        sig_dim = Shape(tile_slice.origin, sig_dims=2).sig
+        reshaped_origin = tuple([nav_dim]) + sig_dim
+        reshaped_tile_slice_shape = Shape(tile_slice.shape, sig_dims=2).flatten_nav()
+        reshaped_tile_slice = Slice(shape=reshaped_tile_slice_shape, origin=reshaped_origin)
+        return reshaped_tile_slice
+
+    def _get_tile_slice_with_positive_offset(self, tile_slice, buf_shape):
```

Comment:
Just a note: this is quite a bit of code, which is executed for each tile. One has to be a bit careful about performance in `get_tiles`, compared to when working on a per-partition level (example: tiles have sizes on the order of 1MiB, while partitions are more like 256MiB to 1GiB, so anything you do in the `get_tiles` codepath gets executed up to 1000 times more!).

Don't worry about it right now, but once #819 lands, there should be a benchmark measuring the overhead of applying an offset